build.sc

```diff
@@ -6,147 +6,164 @@ import coursier.maven.MavenRepository
+        //commonScalacOptions
+
+      override def testFrameworks =
+        Seq("org.scalatest.tools.Framework")
 
-    override def moduleDeps: Seq[JavaModule] =
-      Seq(scalanet)
+      override def ivyDeps = Agg(
+        ivy"org.scalatest::scalatest:3.0.5",
+        ivy"org.scalacheck::scalacheck:1.14.0",
+        ivy"ch.qos.logback:logback-core:1.2.3",
+        ivy"ch.qos.logback:logback-classic:1.2.3",
+        ivy"org.mockito:mockito-core:2.21.0"
+      )
+
+      override def moduleDeps: Seq[JavaModule] =
+        Seq(scalanet)
 
-    def single(args: String*) = T.command {
-      super.runMain("org.scalatest.run", args: _*)
+      def single(args: String*) = T.command {
+        super.runMain("org.scalatest.run", args: _*)
+      }
     }
   }
-}
 
-// In objects inheriting this trait, use `override def moduleDeps: Seq[PublishModule]`
-// to point at other modules that also get published. In other cases such as tests
-// it can be `override def moduleDeps: Seq[JavaModule]`.
-trait ScalanetPublishModule extends PublishModule {
-  def description: String
-
-  override def publishVersion = "0.5.0-SNAPSHOT"
-
-  override def pomSettings = PomSettings(
-    description = description,
-    organization = "io.iohk",
-    url = "https://github.com/input-output-hk/scalanet",
-    licenses = Seq(License.`Apache-2.0`),
-    versionControl = VersionControl.github("input-output-hk", "scalanet"),
-    developers = Seq()
-  )
-}
 
-// ScoverageModule creates bug when using custom repositories:
-// https://github.com/lihaoyi/mill/issues/620
-//object scalanet extends ScalaModule with PublishModule with ScoverageModule {
-object scalanet extends ScalanetModule with ScalanetPublishModule {
-
-  override val description =
-    "Asynchronous, strongly typed, resource-managed networking library, written in Scala with support for a variety of network technologies"
-
-  override def ivyDeps = Agg(
-    ivy"io.monix::monix:3.2.2",
-    ivy"com.github.nscala-time::nscala-time:2.22.0",
-    ivy"com.chuusai::shapeless:2.3.3",
-    ivy"com.typesafe.scala-logging::scala-logging:3.9.2",
-    ivy"com.github.jgonian:commons-ip-math:1.32",
-    ivy"org.slf4j:slf4j-api:1.7.25",
-    ivy"io.netty:netty-all:4.1.51.Final",
-    ivy"org.eclipse.californium:scandium:2.0.0-M15",
-    ivy"org.eclipse.californium:element-connector:2.0.0-M15",
-    ivy"org.scodec::scodec-bits:1.1.12",
-    ivy"org.scodec::scodec-core:1.11.7",
-    ivy"org.bouncycastle:bcprov-jdk15on:1.64",
-    ivy"org.bouncycastle:bcpkix-jdk15on:1.64",
-    ivy"org.bouncycastle:bctls-jdk15on:1.64"
-  )
+  // In objects inheriting this trait, use `override def moduleDeps: Seq[PublishModule]`
+  // to point at other modules that also get published. In other cases such as tests
+  // it can be `override def moduleDeps: Seq[JavaModule]`.
+  trait ScalanetPublishModule extends PublishModule {
+    def description: String
 
-  def scoverageVersion = "1.3.1"
+    override def publishVersion = "0.5.0-SNAPSHOT"
```

Comment:
```suggestion
    override def publishVersion = "0.5.1-SNAPSHOT"
```
It's a bit of a pain but once we start using a version in a project which uses Nix expressions we have to increment the version here otherwise all builds in those projects will constantly break down because of hash changes. Konrad is already migrating to 0.5.0 so we should bump it. 

I guess this will be difficult to coordinate, so we should have explicit release process which removes the `-SNAPSHOT`, and projects that _do_ want to compile against SNAPSHOT should deal with the Nix expressions. @KonradStaniec how do you do releases?

Reply:
0.5.1 is now officially set :1st_place_medal:

Reply:
@aakoshh to this time i have only done release one time so the process is a bit ad hoc and manual:
1.  create release branch
2. create commit which removes `-SNAPHOST` version, and add necessary modification to build.sc (last time I needed to modify it add myslef do `developers` field to by able to do realease). In general we do not have mill version, and thigs change based on mill version, which breaks things pretty often. (maybe we should fix mill version to make it easier, wdyt ?)
3. tag this commit
4. run mill incantation i.e the same thing which is running when merging to develop, but with `--relase` flag

Reply:
I see we have a `master` branch as well. Would it be enough to merge `develop` into `master`, update the version there, then tag it, and add a pipeline step that publishes with `--release` if the tag looks like a semver? As it's described in the [Gitflow workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) I'd only use release branches as a short lived branch where any remaining bugs are sorted out before it's merged into `master` and `develop`, then get rid of them. That would avoid changes lingering there forever, not making it into the next release, etc, and also any ambiguity of whether a specific version should be looked for as a branch or as a tag.

Reply:
I also used to have `build.sbt` take the `version` from the git tag, if it looked like a semver, so we don't have to remove the `SNAPSHOT` and then add it back when the `master` or the `release` is merged back into `develop`.